back/app/services/surveys/results_generator.rb

```diff
@@ -340,6 +372,35 @@ def add_question_numbers_to_results(results)
       end
     end
 
+    def add_averages_for_previous_period(results, start_month, end_month)
+      return results unless start_month && end_month
+
+      # Reduce the fields only to the ones that are linear scales
+      linear_scale_fields = fields.select(&:supports_linear_scale?)
+
+      # Reset inputs to all inputs
+      @inputs = phase.ideas.native_survey.published
+
+      # Find the dates for the previous period
+      start_date = Date.parse("#{start_month}-01")
+      end_date = Date.parse("#{end_month}-01") >> 1 # Start of the next month
+      months_between = ((end_date.year * 12) + end_date.month) - ((start_date.year * 12) + start_date.month)
+      previous_start_month = (start_date << months_between).strftime('%Y-%m')
+      previous_end_month = (end_date << (months_between + 1)).strftime('%Y-%m')
```

Comment:
@jamesspeake Maybe one suggestion here. I was a bit confused about why `months_between + 1` was needed at first, but then I realized it's needed because we first do this a couple lines above: 

`end_date = Date.parse("#{end_month}-01") >> 1 # Start of the next month`

Not sure if a comment could help explain? Maybe it's just me though so feel free to ignore if you think it's clear as-is! :blush:

Reply:
You're right, it's not clear why we have to take it away and then add it back in! Actually, I can see a more obvious way of doing it to get the correct number of number of months now. Will update